### ISSUE TITLE
Is ORT supporting custom backward functions in PyTorch?

### ISSUE THREAD

**Issue by mamesmak** [microsoft/onnxruntime#4370]
We have a Bert model with custom backward. It runs like this in PyTorch:

loss = model(inputs)
model_grads = model.backward(inputs)

(native pytorch is calling loss.backward() which is wrapped in ORTTrainer if I understand correctly)

Wondering if it is possible to run it with ORT?

**Comment by SherlockNoMad** [comment]
ORTTrainer is the user interface to interact with. 
It accepts a model as the input and returns a trainer object.  `trainer = ORTTrainer(model, ...) `
`trainer.train_step(inputs, label, ...)` will do the forward + backward +  weight_update all together. 

Under the hood, ORT builds its own gradient graph and runs forward and backward as one graph. 
I hope this answers your question.

**Comment by mamesmak** [comment]
Thanks Sherlock.
What if the model has it's own backward function? Is ORT able to handle the model.backward() instead of PyTorch's native loss.backward()?

**Comment by SherlockNoMad** [comment]
ORT trainer has its own API for calling backward(). So neither model.backward nor loss.backward would be needed here... 
Is there any special reason why you need model.backward()?

**Comment by mamesmak** [comment]
Yes I have a custom backward which is called by model.backward(inputs) and I need to convert it to ORT. Do you have any suggestions?
Thanks.
